spec/models/preview_card_spec.rb

```diff
@@ -1,5 +1,30 @@
 require 'rails_helper'
 
 RSpec.describe PreviewCard, type: :model do
+  describe '#save_with_optional_image!' do
+    let(:preview_card) { Fabricate(:preview_card) }
 
+    context 'object is valid' do
+      it 'keeps image' do
+        preview_card.save_with_optional_image!
+        expect(preview_card.image.present?).to be true
+      end
+    end
+
+    context 'object is invalid' do
+      before do
+        c = 0
```

Comment:
Since `#save_with_optional_image!` calls `#save!` twice, I don't want to raise `ActiveRecord::RecordInvalid` in second time.